processor/lsmintervalprocessor/internal/merger/model.go

```diff
@@ -406,3 +416,120 @@ func mergeDeltaHistogramDP(from, to pmetric.HistogramDataPoint) {
 		toCounts.SetAt(i, fromCounts.At(i)+toCounts.At(i))
 	}
 }
+
+func mergeDeltaExponentialHistogramDP(from, to pmetric.ExponentialHistogramDataPoint) {
+	if from.Count() == 0 {
+		return
+	}
+	if to.Count() == 0 {
+		from.CopyTo(to)
+		return
+	}
+
+	// Since min/max are optional, set them only when both have it, else reset it
+	if to.HasMin() && from.HasMin() {
+		to.SetMin(min(to.Min(), from.Min()))
+	} else {
+		to.RemoveMin()
+	}
+	if to.HasMax() && from.HasMax() {
+		to.SetMax(max(to.Max(), from.Max()))
+	} else {
+		to.RemoveMax()
+	}
+	to.SetSum(to.Sum() + from.Sum())
+	to.SetCount(to.Count() + from.Count())
+	// TODO (lahsivjar): Zero count depends on the value of zero threshold.
+	// If zero threshold are not the same then we can't merge them directly,
+	// instead, we will need to expand the zero threshold to be inclusive
+	// of the smaller zero count. If a zero threshold encroaches a bucket,
+	// then we will need to expand the zero count till the end of the bucket.
+	to.SetZeroCount(to.ZeroCount() + from.ZeroCount())
+
+	// Downscale the resolution and use perfect-subsetting
+	minScale := min(to.Scale(), from.Scale())
+	maybeScaleDownHistogram(to, minScale)
+	maybeScaleDownHistogram(from, minScale)
+
+	// All histogram buckets are now on same scale, merge them
+	mergeExponentialHistogramBuckets(from.Positive(), to.Positive())
+	mergeExponentialHistogramBuckets(from.Negative(), to.Negative())
+}
+
+func mergeExponentialHistogramBuckets(from, to pmetric.ExponentialHistogramDataPointBuckets) {
+	fromBuckets := from.BucketCounts()
+	toBuckets := to.BucketCounts()
+	fromSize := int(from.Offset()) + fromBuckets.Len()
+	toSize := int(to.Offset()) + toBuckets.Len()
+
+	if to.Offset() > from.Offset() {
+		// Expand `to` to accomodate the merged buckets with offset as `from.Offset()`
+		newBucketsSize := max(fromSize, toSize) - int(from.Offset())
+		newBuckets := pcommon.NewUInt64Slice()
+		newBuckets.Append(make([]uint64, newBucketsSize)...)
+
+		// Copy `to` to the new bucket
+		for i := 0; i < toBuckets.Len(); i++ {
+			newBuckets.SetAt(int(to.Offset())+i, toBuckets.At(i))
```

Comment:
how will this work when both to and from offsets are negative?

Reply:
I don't think offsets can be negative, they are number of zeroes and negative buckets are separated. Anway, check this https://github.com/elastic/opentelemetry-collector-components/pull/151#discussion_r1783675061 (TL;DR, we are vendoring an upstream internal implementation)

Reply:
For posterity, Carson is correct here. The offset cannot be taken directly to calculate indexes, they must be normalized first by taking the difference of max of the upper bounds and min of the lower bounds.